### ISSUE TITLE
About send ack to client with Socket.io

### ISSUE THREAD

**Issue by Frankzhaopku** [balderdashy/sails#2960]
I use sails socket as my chat server and I need to send an ack message back to client when I receive one.

On socket.io site, there is a demo to send an ack at server end like below:
io.on('connection', function (socket) {
  socket.on('ferret', function (name, fn) {
    fn('woot');
  });
});

But how can I use the fn callback in sails MVC framework?

Thx.

**Comment by Frankzhaopku** [comment]
I found a file sails-hook-sockets/lib/on-connect.js and send back an ack message by socketIOClientCallback in receiveMessage function, is there any access to this function at the upper layer of sails?

**Comment by sailsbot** [comment]
Thanks for posting, @Frankzhaopku.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
